### PR TITLE
Extract permissions union of all role assignments

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-1057

### Change description ###

Extract permissions union of all role assignments, minor refactoring for better code split, minor optimizations

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by dharmendrak — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetTaskByIdControllerCFTTest.java [hunk 1]
```diff
@@ -243,7 +243,7 @@ public void should_return_a_200_with_task_warnings() {
             .and().contentType(MediaType.APPLICATION_JSON_VALUE)
             .and().body("task.id", equalTo(taskId))
             .body("task.warnings", is(true))
-            .body("task.permissions.values", hasItems("Read","Refer","Own","Manage","Cancel"));
+            .body("task.permissions.values", hasItems("Read","Refer","Own"));
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetTaskByIdControllerCFTTest.java:246]: as per requirement, we will need to change the matcher to check this contains only the permissions relevant to the user

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetTaskByIdControllerCFTTest.java [hunk 2]
```diff
@@ -304,7 +304,7 @@ public void should_return_a_200_with_task_and_correct_properties() {
             .body("task.case_name", notNullValue())
             .body("task.auto_assigned", notNullValue())
             .body("task.warnings", notNullValue())
-            .body("task.permissions.values", hasItems("Read", "Refer", "Own", "Manage", "Cancel"))
+            .body("task.permissions.values", hasItems("Read", "Refer", "Own"))
             .body("task.description", notNullValue())
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetTaskByIdControllerCFTTest.java:308]: as per requirement, we will need to change the matcher to check this contains only the permissions relevant to the user

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java [hunk 3]
```diff
@@ -102,7 +102,7 @@ public void should_return_200_with_appropriate_task_to_complete() {
                 .statusCode(HttpStatus.OK.value())
                 .contentType(APPLICATION_JSON_VALUE)
                 .body("tasks.size()", equalTo(1))
-                .body("tasks[0].permissions.values", hasItems("Read","Refer","Own","Manage","Cancel"))
+                .body("tasks[0].permissions.values", hasItems("Read","Refer","Own"))
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java:105]: as per requirement, we will need to change the matcher to check this contains only the permissions relevant to the user

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java [hunk 4]
```diff
@@ -260,7 +260,7 @@ public void should_return_a_200_and_retrieve_a_task_by_event_and_case_match() {
             .body("tasks[0].type", equalTo("reviewTheAppeal"))
             .body("tasks[0].jurisdiction", equalTo("IA"))
             .body("tasks[0].case_type_id", equalTo("Asylum"))
-            .body("tasks[0].permissions.values", hasItems("Read", "Refer", "Own", "Manage", "Cancel"))
+            .body("tasks[0].permissions.values", hasItems("Read", "Refer", "Own"))
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java:263]: as per requirement, we will need to change the matcher to check this contains only the permissions relevant to the user

**Review by BenNzewi — COMMENTED** [review]
We also need to update the tests in  /task endpoint (Search)

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java (hunk 4 above)
Comment (alexbaeza) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerCFTTest.java:266]: Added as part of new commit

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**